lib/register_sources_bods/register/entity_service.rb

```diff
@@ -60,29 +60,32 @@ def count_legal_entities
       # merged_page, source_page
 
       def find(statement_id)
-        result = statement_loader.load_statements([statement_id])
-
+        resolved_statement_id = statement_id.split('-unknown').first
+        result = statement_loader.load_statements([resolved_statement_id])
         result.entities[statement_id] || result.relationships[statement_id]
       end
 
       def find_by_entity_id(entity_id)
         find_by_entity_uri("/entities/#{entity_id}")
       end
 
-      def find_by_entity_ids(entity_ids)
-        find_by_entity_uris(entity_ids.map { |entity_id| "/entities/#{entity_id}" })
-      end
-
       def find_by_entity_uri(uri)
-        find_by_entity_uris([uri]).first
+        results = find_by_entity_uris([uri])
+
+        if /unknown/.match uri
```

Comment:
Or `results.find { |result| (/unknown/.match(uri))(/unknown/.match(result.id)) }` or something similar (the precise logic could be wrong). I suspect it might be able to be simplified further, though… And the logic confuses me a bit.